keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationSupport.kt

```diff
@@ -18,59 +18,76 @@
 package com.netflix.spinnaker.keel.rest
 
 import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator
-import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.api.application
 import com.netflix.spinnaker.keel.api.serviceAccount
+import com.netflix.spinnaker.keel.exceptions.NoSuchEntityException
 import com.netflix.spinnaker.keel.persistence.KeelRepository
-import com.netflix.spinnaker.keel.persistence.NoSuchResourceException
+import com.netflix.spinnaker.keel.rest.AuthorizationSupport.Entity.APPLICATION
+import com.netflix.spinnaker.keel.rest.AuthorizationSupport.Entity.DELIVERY_CONFIG
+import com.netflix.spinnaker.keel.rest.AuthorizationSupport.Entity.RESOURCE
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import org.springframework.security.core.Authentication
 import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.stereotype.Component
 
+/**
+ * Support for authorization of REST API calls.
+ *
+ * Permission is granted based on the user (as identified by the X-SPINNAKER-USER request header, or X509 client
+ * certificate, or metatron identity) having access to the "service account" associated with keel's entities
```

Comment:
Non-Netflixers likely won't know what "metatron" refers to, so may want to note that it's a Netflix-internal credential management service.